### DISCUSSION TITLE
Current ECUs back up

### DISCUSSION THREAD

**Discussion by Trickykid1978** [cedricp/ddt4all#986] [Q&A]
Hello everybody,  can someone explain me the process of backing up and restore (if needed)  car's current ECUs  ?  If i save these ecus in the ecu folder of ddt4all and rescan, then these ecus will always appear in Green color ?

**Comment by AndyIreland** [comment]
try use PyREN instead to make a backup / Dump of your ECU's its much better for that 
https://www.drive2.ru/b/501493851572142178/

**Reply by BadFoxer** [reply]
Andy can you tell me how virginize eps form modus in PyREN ? thanks

**Reply by AndyIreland** [reply]
I cannot I am sorry because I have never performed virginize eps before. 
You should make a backup of your ECU though here at the very least: 
![image](https://github.com/cedricp/ddt4all/assets/49172217/1d030090-d415-42a0-bac0-28a99647c02d)

then 
![image](https://github.com/cedricp/ddt4all/assets/49172217/ada0c29d-1823-47bc-9ee5-ecdc455b137e)

**Reply by BadFoxer** [reply]
Maybe i can try do with ddt4all i found eps virginize ? i have elm327

![294050104-e83b6c24-642a-4952-a643-01a7041d4390](https://github.com/cedricp/ddt4all/assets/31113326/9b554fb3-1b05-404b-9487-ad2e93a60bcd)

**Reply by KhalidL987** [reply]
Can you please help me by explaining the steps to make a backup of all ECU before running a macro?
